### PR TITLE
feat(gateway): human error pages, dag-cbor/dag-json preview

### PR DESCRIPTION
**Pull request by hacdias**
- [x] Changelog Entry
- [x] https://github.com/ipfs/boxo/pull/298
- [x] https://github.com/ipfs/boxo/pull/315

### CODE REVIEW

**Review by lidel — APPROVED** [review]
Thank you @hacdias, this is a significant quality of life improvement for gateway users, and allows us to improve it even further in follow up PRs.

Merging to ensure this is included in Kubo 0.21.0-rc1 (https://github.com/ipfs/kubo/issues/9814), and we can get feedback from stakeholders like @anjor.

File: docs/changelogs/v0.21.md [hunk 1]
```diff
@@ -59,7 +60,7 @@ third-party CIDs.
 #### `client/rpc` migration of `go-ipfs-http-client`
 
 The [`go-ipfs-http-client`](https://github.com/ipfs/go-ipfs-http-client) RPC has
-been migrated into [`client/rpc`](./client/rpc).
+been migrated into [`kubo/client/rpc`](https://github.com/ipfs/kubo/tree/master/client/rpc).
```
Comment (Jorropo) [inline docs/changelogs/v0.21.md:63]:
```suggestion
been migrated into [`kubo/client/rpc`](../client/rpc).
```
mb

**Review by Jorropo — DISMISSED** [review]

File: docs/changelogs/v0.21.md (hunk 1 above)
Comment (lidel) [reply docs/changelogs/v0.21.md:63]: @Jorropo  `/docs/client/rpc`  does not seem to exist?

Comment (Jorropo) [reply docs/changelogs/v0.21.md:63]: `../..`

**Review by Jorropo — APPROVED** [review]

**Pull request merged**